src/redis.c

```diff
@@ -426,6 +435,12 @@ static int redis_read(void) /* {{{ */
       freeReplyObject(rr);
     }
 
+    if ((rr = redisCommand(rh, "SELECT %d", rn->db)) == NULL) {
```

Comment:
I think we have to skip this if `rn->db == 0`. 
https://redis.io/commands/select explicitly says what new connections always use the database 0 and we always have new connections here.
What do you think?